clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3149,6 +3147,56 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
     KernelFunc->setInvalidDecl();
 }
 
+// For a wrapped parallel_for, copy attributes from original
+// kernel to wrapped kernel.
+void Sema::copySYCLKernelAttrs(const CXXRecordDecl *KernelObj) {
```

Comment:
This function should not take a `const CXXRecordDecl *` when it's mutating the state of that object's methods (it's fine as far as the C++ type system is concerned but is logically questionable given that we're mutating the state of the object).

Reply:
Removing the const here has a cascading effect because of the large number of interdependent function calls.
The pointer variable with the const is not modified in the function so it is correct to call it const. If we were claiming not to modify what this pointer points to the declaration would have been const CXXRecordDecl *const .. but it isn't.

Reply:
I believe you that there are cascading effects, but I don't agree that it's correct to call it a `const CXXRecordDecl`. Clang's code base is pretty bad at const correctness (it's a best-faith effort), so these cascading effects tell me to not bother trying to change the signature. Thanks for checking, I think it's fine as-is.